pkgs/build-support/zig/build-zig-package/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, zig, mktemp }:
+
+{ strictDeps ? true
+, nativeBuildInputs ? [ ]
+, enableParallelBuilding ? true
+, meta ? { }
+
+, zigDefaultBuildFlags ? [ "-Drelease-safe" "-Dcpu=baseline" ]
+, zigBuildFlags ? [ "" ]
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
+  inherit strictDeps enableParallelBuilding;
+  nativeBuildInputs = [ zig ] ++ nativeBuildInputs;
+
+  postConfigure = attrs.postConfigure or ''
+    export HOME=$(mktemp -d)
+  '';
+
+  buildPhase = attrs.buildPhase or ''
+    runHook preBuild
+    zig build ${lib.concatStringsSep " " (zigDefaultBuildFlags ++ zigBuildFlags)}
```

Comment:
```suggestion
    zig build -Drelease-safe -Dcpu=baseline $zigBuildFlags
```

Reply:
I do not like this idea of hardcoded flags.

Reply:
[this](https://github.com/NixOS/nixpkgs/pull/241741#issuecomment-1623970983) is my reasoning, I think we should hold off on making this customizable to avoid making breaking changes in 10 days